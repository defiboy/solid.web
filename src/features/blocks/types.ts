import { Transaction } from "features/transactions";

export interface BlockHeader {
    number: number
    hash: string
    parentHash: string
    nonce: string
    sha3Uncles: string
    logsBloom: string
    transactionRoot: string
    stateRoot: string
    receiptRoot: string
    miner: string
    extraData: string
    gasLimit: number
    gasUsed: number
    timestamp: number | string
}

export interface Block extends BlockHeader {
    transactions: Transaction[];
    size: number
    difficulty: number
    totalDifficulty: number
    uncles: string[];
}