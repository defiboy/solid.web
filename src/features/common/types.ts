export enum Status { // TODO, with sagas I think it won't apply
    InProgress = 'InProgress',
    Completed = 'Completed',
    Synchronizing = 'Synchronizing', // this won't apply for connection modal
    Failed = 'Failed',
    NotStarted = 'NotStarted',
    Started = 'Started'
}

export interface GenericResponse<T> {
    total: number;
    limit: number;
    skip: number;
    data: T;
}

export interface GenericArrayResponse<T> {
    total: number;
    limit: number;
    skip: number;
    data: T[];
}
