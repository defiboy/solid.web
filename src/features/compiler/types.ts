import { Status } from '../common/types'

export interface ValidateSourceCode {
  status: Status
  compilerVersion: string
  sourceCode: string
}

export interface LoadCompilerRequest {
  version: string
  status: Status
}
