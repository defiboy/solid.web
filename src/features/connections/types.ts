import { Connection } from '@solid-explorer/types'

export interface ConnectionItem extends Connection {
  type: string
}
