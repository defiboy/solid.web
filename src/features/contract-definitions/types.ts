import { ContractDefinition } from '@solidstudio/solid.types'

export interface ContractDefinitionItem extends ContractDefinition {
    type: string
}