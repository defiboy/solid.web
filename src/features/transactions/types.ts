export interface Transaction {
    // TODO
    status: boolean
    transactionHash: string
    transactionIndex: number
    blockHash: string
    blockNumber: number
    from: string
    to: string
    contractAddress?: string
    cumulativeGasUsed: number
    gasUsed: number
}