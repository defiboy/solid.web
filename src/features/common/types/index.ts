export enum Status { // TODO, with sagas I think it won't apply
    InProgress = 'InProgress',
    Completed = 'Completed',
    Synchronizing = 'Synchronizing', // this won't apply for connection modal
    Failed = 'Failed',
    NotStarted = 'NotStarted',
    Started = 'Started'
}