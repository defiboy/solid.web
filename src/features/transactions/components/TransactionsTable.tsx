import React from 'react'

import { Tag, Button } from 'antd';
import { ColumnProps } from 'antd/es/table';

import { TransactionReceipt } from '@solid-explorer/types'

import { TransactionsTableComponent } from './TransactionsTableComponent';
import styled from 'styled-components';

interface OwnProps {
    transactions?: TransactionReceipt[]
    onClick: (record: TransactionReceipt) => void
    onDoubleClick: (record: TransactionReceipt) => void
}

type AllProps = OwnProps // & DispatchProps & StateProps

const failedTransactionTag = (key: string) => (
    <Tag color="volcano" key={key}>
        Failed
    </Tag>
)

const successfulTransactionTag = (key: string) => (
    <Tag color="green" key={key}>
        Success
    </Tag>
)

const contractCreationTag = (key: string) => (
    <Tag color="geekblue" key={key}>
        Contract Creation
    </Tag>
)

const contractCallTag = (key: string) => (
    <Tag color="geekblue" key={key}>
        Contract Call
    </Tag>
)

// TODO Move this to another place
const DebugButton = styled(Button)`
    padding: 0 10px;
    font-size: 12px;
`

const tableColumns: Array<ColumnProps<TransactionReceipt>> = [
    // {
    //     key: 'type',
    //     title: 'Type',
    //     dataIndex: 'type',
    //     render: (text: string, record: Transaction) => record.to === null ? contractCreationTag(record.transactionHash) : contractCallTag(record.transactionHash)
    // },
    {
        key: 'transactionHash',
        title: 'Transaction Hash',
        dataIndex: 'transactionHash',
        // width: 5,
        // align: 'center',
        render: text => <p data-testid={`transactions-table-row-${text}`}>{text}</p>
    },
    {
        key: 'to',
        title: 'To',
        dataIndex: 'to',
        // align: 'center',
        render: (text: string, record: TransactionReceipt) => record.to == null ? contractCreationTag(record.transactionHash) : record.to
    },
    {
        key: 'from',
        title: 'From',
        dataIndex: 'from',
        // align: 'center',

    },
    {
        key: 'contractAddress',
        title: 'Contract Address',
        dataIndex: 'contractAddress',
    },
    // {
    //     key: 'status',
    //     title: 'Status',
    //     dataIndex: 'status',
    //     render: (text: string, record: TransactionReceipt) => record.status ? failedTransactionTag(record.transactionHash) : failedTransactionTag(record.transactionHash)
    // },
    {
        title: 'Actions',
        key: 'action',
        render: (text, record) => (
            <DebugButton type="danger" size="small">Debug</DebugButton>
        )
    }
    // {
    //     key: 'executionDate',
    //     title: 'Execution Date',
    //     dataIndex: 'executionDate',
    // }

];

export class TransactionsTable extends React.Component<AllProps> {
    render() {
        const { transactions } = this.props
        return <TransactionsTableComponent
            rowKey="transactionHash"
            dataSource={transactions}
            columns={tableColumns}
            onRow={(record, rowIndex) => {
                return {
                    onClick: event => {
                        this.props.onClick(record)
                    },
                    onDoubleClick: event => {
                        this.props.onDoubleClick(record)
                    }
                };
            }}
        />
    }
}