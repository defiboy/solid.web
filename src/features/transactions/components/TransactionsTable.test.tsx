import React from 'react';
import { render, fireEvent } from '@testing-library/react'

import { TransactionReceipt, buildFakeTransactionReceipts } from '@solid-explorer/types'

import { TransactionsTable } from './TransactionsTable'

describe('Transactions table test', () => {

    let transactions: TransactionReceipt[]
    const mockOnClickTable = jest.fn()
    const mockOnDoubleClickTable = jest.fn()

    const renderTransactionsTable = (transactions: TransactionReceipt[] | undefined) => {
        return render(
            <TransactionsTable transactions={transactions} onClick={mockOnClickTable} onDoubleClick={mockOnDoubleClickTable} />
        )
    }

    beforeEach(() => {
        transactions = buildFakeTransactionReceipts()
        mockOnDoubleClickTable.mockReset()
        mockOnClickTable.mockReset()
    })

    test('that transactions table renders correctly with data', () => {
        const { getByTestId } = renderTransactionsTable(transactions)

        transactions.forEach((item) => {
            const row = getByTestId(`transactions-table-row-${item.transactionHash}`)
            expect(row).toBeInTheDocument()
        })
    })

    test('that table renders correctly without data', () => {
        const { getByText } = renderTransactionsTable(undefined)

        const noDataInTable = getByText(/no data/i)

        expect(noDataInTable).toBeInTheDocument()
    })

    test('snapshot', () => {
        const { container } = renderTransactionsTable(transactions)

        expect(container).toMatchSnapshot()
    })

    test('clicks', () => {
        const { getByTestId } = renderTransactionsTable(transactions)

        transactions.forEach((item) => {
            const row = getByTestId(`transactions-table-row-${item.transactionHash}`)

            fireEvent.click(row)

            expect(mockOnClickTable).toHaveBeenCalledTimes(1)
            expect(mockOnClickTable).toHaveBeenCalledWith(item)
            expect(mockOnDoubleClickTable).toHaveBeenCalledTimes(0)
            mockOnClickTable.mockClear()
        })
    })

    test('double clicks', () => {
        const { getByTestId } = renderTransactionsTable(transactions)

        transactions.forEach((item) => {
            const row = getByTestId(`transactions-table-row-${item.transactionHash}`)

            fireEvent.doubleClick(row)

            expect(mockOnDoubleClickTable).toHaveBeenCalledTimes(1)
            expect(mockOnDoubleClickTable).toHaveBeenCalledWith(item)
            expect(mockOnClickTable).toHaveBeenCalledTimes(0)
            mockOnDoubleClickTable.mockClear()
        })
    })
})