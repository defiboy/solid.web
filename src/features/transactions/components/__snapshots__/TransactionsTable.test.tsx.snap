// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions table test snapshot 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-default ant-table-scroll-position-left"
        >
          <div
            class="ant-table-content"
          >
            <div
              class="ant-table-body"
            >
              <table
                class=""
              >
                <colgroup>
                  <col />
                  <col />
                  <col />
                  <col />
                  <col />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Transaction Hash
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            To
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            From
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Contract Address
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Status
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="0xdf7f4433c421ab496a15066c14188b3e1973591cf539a9bd55009f108fddb668"
                  >
                    <td
                      class=""
                    >
                      <p
                        data-testid="transactions-table-row-0xdf7f4433c421ab496a15066c14188b3e1973591cf539a9bd55009f108fddb668"
                      >
                        0xdf7f4433c421ab496a15066c14188b3e1973591cf539a9bd55009f108fddb668
                      </p>
                    </td>
                    <td
                      class=""
                    >
                      <span
                        class="ant-tag ant-tag-geekblue"
                      >
                        Contract Creation
                      </span>
                    </td>
                    <td
                      class=""
                    >
                      0x97a4a7b888d9ffa3a4e4c5a407bb557a6a2b7be8
                    </td>
                    <td
                      class=""
                    >
                      0xb5a04884b36984a4d27a796e7fa75561d6a5533a
                    </td>
                    <td
                      class=""
                    >
                      <span
                        class="ant-tag ant-tag-green"
                      >
                        Success
                      </span>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="0xdf7f4433c421ab496a15066c14188b3e1973591cf539a9bd55009f108fddb667"
                  >
                    <td
                      class=""
                    >
                      <p
                        data-testid="transactions-table-row-0xdf7f4433c421ab496a15066c14188b3e1973591cf539a9bd55009f108fddb667"
                      >
                        0xdf7f4433c421ab496a15066c14188b3e1973591cf539a9bd55009f108fddb667
                      </p>
                    </td>
                    <td
                      class=""
                    >
                      <span
                        class="ant-tag ant-tag-geekblue"
                      >
                        Contract Creation
                      </span>
                    </td>
                    <td
                      class=""
                    >
                      0x97a4a7b888d9ffa3a4e4c5a407bb557a6a2b7be8
                    </td>
                    <td
                      class=""
                    >
                      0xb5a04884b36984a4d27a796e7fa75561d6a5533a
                    </td>
                    <td
                      class=""
                    >
                      <span
                        class="ant-tag ant-tag-green"
                      >
                        Success
                      </span>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination ant-table-pagination"
          unselectable="unselectable"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-prev"
            title="Previous Page"
          >
            <a
              class="ant-pagination-item-link"
            >
              <i
                aria-label="icon: left"
                class="anticon anticon-left"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </i>
            </a>
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-next"
            title="Next Page"
          >
            <a
              class="ant-pagination-item-link"
            >
              <i
                aria-label="icon: right"
                class="anticon anticon-right"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                  />
                </svg>
              </i>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
